### PR TITLE
Hook tutorial

### PR DESCRIPTION
**Pull request by skyerus**

### CODE REVIEW

File: blog/26-08-2022-creating-a-hook-for-the-go-sdk.mdx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+A [Hook](https://docs.openfeature.dev/docs/specification/sections/hooks) taps into one or more of the flag evaluation's
+lifecycle events (before/after/error/finally) to perform the same action at that point for every evaluation.
+
+## Objective
+
+<i>Create and integrate a spec compliant hook that validates that the result of a flag evaluation is a hex color.</i>
+
+## Prerequisites
+
+- Golang 1.17
```
Comment (beeme1mr) [inline blog/26-08-2022-creating-a-hook-for-the-go-sdk.mdx:20]: Is the min version 1.17 or is it the required version?

**Review by beeme1mr — APPROVED** [review]
Looks good to me, thanks!

File: blog/26-08-2022-creating-a-hook-for-the-go-sdk.mdx (hunk 1 above)
Comment (toddbaert) [reply blog/26-08-2022-creating-a-hook-for-the-go-sdk.mdx:20]: Ya, I think this should say `Golang 1.17+`

**Review by toddbaert — APPROVED** [review]
This looks really great. Well written and explained and to the point. Thanks!

File: blog/26-08-2022-creating-a-hook-for-the-go-sdk.mdx (hunk 1 above)
Comment (skyerus) [reply blog/26-08-2022-creating-a-hook-for-the-go-sdk.mdx]: Nice spot, done!

File: sidebars.js [hunk 2]
```diff
@@ -36,11 +36,8 @@ const sidebars = {
   ],
   tutorials: [
     {
-      type: 'category',
-      label: 'Tutorials',
-      link: { type: 'doc', id: 'tutorials/overview' },
-      collapsible: false,
-      items: [],
+      type: 'autogenerated',
+      dirName: 'tutorials',
```
Comment (beeme1mr) [inline sidebars.js:40]: Is this used anymore?

File: blog/26-08-2022-creating-a-hook-for-the-go-sdk.mdx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+slug: creating-a-hook-for-the-go-sdk
+title: Creating a Hook for the golang-sdk
+authors: [skyerus]
+tags: [golang, blog, tutorial, golang-sdk, hooks]
+unlisted: false
+---
+
+## Hooks
+
+A [Hook](https://docs.openfeature.dev/docs/specification/sections/hooks) taps into one or more of the flag evaluation's
```
Comment (beeme1mr) [inline blog/26-08-2022-creating-a-hook-for-the-go-sdk.mdx:11]: Are you able to use just the path or a relative path here?

File: sidebars.js (hunk 2 above)
Comment (skyerus) [reply sidebars.js:40]: It's not but I left it in hope that it'd save someone else some confusion in the future. Happy to remove if you'd prefer

File: blog/26-08-2022-creating-a-hook-for-the-go-sdk.mdx (hunk 3 above)
Comment (skyerus) [reply blog/26-08-2022-creating-a-hook-for-the-go-sdk.mdx]: Done

File: sidebars.js (hunk 2 above)
Comment (beeme1mr) [reply sidebars.js:40]: You can leave it if it was left here intentionally.

File: blog/26-08-2022-creating-a-hook-for-the-go-sdk.mdx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+slug: creating-a-hook-for-the-go-sdk
+title: Creating a Hook for the golang-sdk
+authors: [skyerus]
+tags: [golang, blog, tutorial, golang-sdk, hooks]
+unlisted: false
+---
+
+## Hooks
+
+A [Hook](./../docs/specification/sections/hooks) taps into one or more of the flag evaluation's
```
Comment (beeme1mr) [inline blog/26-08-2022-creating-a-hook-for-the-go-sdk.mdx:11]: Apparently you can't do relative paths in blog posts.. bummer.

Comment (skyerus) [reply blog/26-08-2022-creating-a-hook-for-the-go-sdk.mdx:11]: Ah that's annoying, it worked locally :(

Comment (skyerus) [reply blog/26-08-2022-creating-a-hook-for-the-go-sdk.mdx:11]: Will revert

**Pull request merged**